utils/symbol.c

```diff
@@ -824,6 +824,10 @@ enum uftrace_trace_type check_trace_functions(const char *filename)
 
 		/* undefined function is ok here */
 		if (elf_symbol_type(&iter.sym) != STT_FUNC &&
+#ifdef __ANDROID__
+		    // Profiling functions are undefined on Android
+		    elf_symbol_type(&iter.sym) != STT_NOTYPE &&
```

Comment:
Executables linked with `-finstrument-functions`/`-pg` need to be linked with `-Wl,-z,undefs` because corresponding callbacks are missing from Android runtime library and only come from `libmcount`.